core/src/Cabana_GaussianMixtureModel.hpp

```diff
@@ -0,0 +1,57 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+/*!
+  \file Cabana_GaussianMixtureModel.hpp
+  \brief Creation of a Gaussian Mixture Model
+*/
+#ifndef CABANA_GMM_HPP
+#define CABANA_GMM_HPP
+
+#include <typeinfo>
+
+// Things to describe a Gaussian
+enum GaussianFields {
+    Weight,
+    // Can we make only the relevant fields vissible based on <dims>?
+    MuPar, MuPer,
+    Cparpar, Cparper,
+    Cperpar, Cperper,
+    MuX, MuY, MuZ,
+    Cxx, Cxy, Cxz,
+    Cyx, Cyy, Cyz,
+    Czx, Czy, Czz,
+    n_gaussian_param
+};
+
+
+#include <impl/Cabana_GaussianMixtureModel.hpp>
```

Comment:
In most of our code we just use one header, but use an `Impl` namespace for anything not intended for the user

Reply:
It's easy enough to change but will clutter up the doxygen page for that file.